fuse/eval/metrics/libs/classification.py

```diff
@@ -61,6 +61,44 @@ def auc_roc(
             y_score=y_score, y_true=np.asarray(target) == pos_class_index, sample_weight=sample_weight, max_fpr=max_fpr
         )
 
+    @staticmethod
+    def auc_roc_mult_label(
+        pred: Sequence[Union[np.ndarray, float]],
+        target: Sequence[Union[np.ndarray, int]],
+        sample_weight: Optional[Sequence[Union[np.ndarray, float]]] = None,
+        average: str = "micro",
+        max_fpr: Optional[float] = None,
+    ) -> float:
+        """
+        Compute multi label auc roc (Receiver operating characteristic) score using sklearn
+        :param pred: prediction array per sample. Each element shape [num_classes]
+        :param target: target per sample. Each element shape [num_classes] with 0 and 1 only.
+        :param sample_weight: Optional - weight per sample for a weighted auc. Each element is  float in range [0-1]
+        :param pos_class_index: the class to compute the metrics in one vs rest manner - set to 1 in binary classification
+        :param max_fpr: float > 0 and <= 1, default=None
+                        If not ``None``, the standardized partial AUC over the range [0, max_fpr] is returned.
+        :return auc Receiver operating characteristic score
+        """
+        y_score = np.asarray(pred)
+        y_true = np.asarray(target)
+        if average == "macro":
+            # compute the roc for each class (column in y_score/y_true) and average
+            n_class = y_score.shape[1]
+            all_auc = []
+            for i in range(n_class):
+                auc = metrics.roc_auc_score(
```

Comment:
We should be careful here and ignore labels that don't have both positive class and negative class